crates/flux-infer/src/fixpoint_encoding.rs

```diff
@@ -1118,8 +1120,9 @@ impl<'genv, 'tcx> ExprEncodingCtxt<'genv, 'tcx> {
         e2: &rty::Expr,
         scx: &mut SortEncodingCtxt,
     ) -> QueryResult<fixpoint::Expr> {
+        // TODO: I understand zero of this
         let e = match sort {
-            rty::Sort::Int | rty::Sort::Real => {
+            rty::Sort::Int | rty::Sort::Real | rty::Sort::Char | rty::Sort::Str => {
```

Comment:
not completely sure if `Str` should be here. our conversation about `bin_rel_to_fixpoint` has somehow completely evaporated from my mind

Reply:
no, strings should not be here because they can't be compared using `<` or `>` in z3. There's a special function for comparing strings see https://microsoft.github.io/z3guide/docs/theories/Strings